src/main/java/org/wildfly/bot/PullRequestFormatProcessor.java

```diff
@@ -146,57 +149,43 @@ private void generateAppendedMessage(GHPullRequest pullRequest, Pattern projectP
         }
     }
 
-    private void appendJiraIssues(StringBuilder sb, GHPullRequest pullRequest, Pattern projectPattern) {
+    private Set<String> getMissingIssues(String originalBody, GHPullRequest pullRequest, Pattern projectPattern) {
         Set<String> jiraIssues = parseJiraIssues(pullRequest, projectPattern);
-        String currentBody = sb.toString();
+        String body = originalBody == null ? "" : originalBody;
         if (jiraIssues.isEmpty()) {
             LOG.debugf("No JIRA issues found for Pull Request [#%s]: \"%s\"", pullRequest.getNumber(), pullRequest.getTitle());
         }
-
         // we create links, search for ones contained in the original body
         Set<String> jiraLinks = jiraIssues.stream()
                 .map(s -> String.format(BOT_JIRA_LINK_TEMPLATE, s))
                 .collect(Collectors.toSet());
 
         // collect back jira issues from links, which are missing in the description
-        jiraIssues = jiraLinks.stream()
-                .filter(s -> !currentBody.contains(s))
+        return jiraLinks.stream()
+                .filter(s -> !body.contains(s))
                 .map(s -> {
                     Matcher matcher = projectPattern.matcher(s);
                     matcher.find();
                     return matcher.group();
                 })
                 .collect(Collectors.toSet());
+    }
 
-        if (jiraIssues.isEmpty()) {
-            return;
-        }
-
-        if (!currentBody.isEmpty()) {
+    private void appendMissingIssues(StringBuilder sb, Set<String> jiraIssues) {
+        if (!sb.isEmpty()) {
             sb.append("\n\n");
         }
 
         sb.append(BOT_MESSAGE_HEADER)
                 .append(blockQuoted(BOT_JIRA_LINKS_HEADER));
 
         for (String jira : jiraIssues) {
-
             sb.append(blockQuoted(String.format(BOT_JIRA_LINK_COMMENT_TEMPLATE, jira)));
         }
 
         sb.append(BOT_MESSAGE_FOOTER);
     }
 
-    private void appendInfoFooter(StringBuilder sb) {
```

Comment:
This removes the footer info all together. I think, we should rather append the footer only in case, we append something else as well. Thus if any Jira issues are appended, then we should also append the footer.
Something like originalBody != newBody -> append footer.